recipes/gdal/post_3.5.0/patches/3.5.1/0-replace-find-package.patch

```diff
@@ -0,0 +1,266 @@
+diff --git a/cmake/helpers/ConanFindPackage.cmake b/cmake/helpers/ConanFindPackage.cmake
+new file mode 100644
+index 0000000000..9dfa8193a3
+--- /dev/null
++++ b/cmake/helpers/ConanFindPackage.cmake
+@@ -0,0 +1,43 @@
++
++function(define_find_package2 pkgname include_file library_name)
++endfunction()
++
++function(find_package2 pkgname)
++    set(_options QUIET REQUIRED)
++    set(_oneValueArgs OUT_DEPENDENCY)
++    set(_multiValueArgs)
++    cmake_parse_arguments(arg "${_options}" "${_oneValueArgs}" "${_multiValueArgs}" ${ARGN})
++    if(arg_QUIET)
++        set(${pkgname}_FIND_QUIETLY TRUE)
++    endif()
++    if(arg_REQUIRED)
++        set(${pkgname}_FIND_REQUIRED TRUE)
++    endif()
++
++    string(TOUPPER ${pkgname} key)
++
++    set(docstring "Configured for conan package ${GDAL_CONAN_PACKAGE_FOR_${key}}")
++    if (DEFINED GDAL_CONAN_PACKAGE_FOR_${key})
++        message("Using conan package ${GDAL_CONAN_PACKAGE_FOR_${key}} for dependency ${pkgname}")
++        set(conan_package ${GDAL_CONAN_PACKAGE_FOR_${key}})
++        string(TOUPPER ${conan_package} conan_package_upper)
++
++        set(${key}_INCLUDE_DIRS "${CONAN_INCLUDE_DIRS_${conan_package_upper}}" CACHE STRING ${docstring})
++        if (NOT TARGET_FOR_${key})
++            set(TARGET_FOR_${key} "${conan_package}::${conan_package}")
++        endif()
++        set(${key}_LIBRARIES "${TARGET_FOR_${key}}" CACHE STRING ${docstring})
++        set(${key}_LIBRARY "${TARGET_FOR_${key}}" CACHE STRING ${docstring})
++        set(${key}_TARGET "${TARGET_FOR_${key}}" CACHE STRING ${docstring})
++        set(${pkgname}_INCLUDE_DIRS "CONAN_INCLUDE_DIRS_${conan_package_upper}" CACHE STRING ${docstring})
++        set(${pkgname}_LIBRARIES "${TARGET_FOR_${key}}" CACHE STRING ${docstring})
++        set(${pkgname}_LIBRARY "${TARGET_FOR_${key}}" CACHE STRING ${docstring})
++        set(${pkgname}_TARGET "${TARGET_FOR_${key}}" CACHE STRING ${docstring})
++        set(${key}_FOUND TRUE CACHE BOOL ${docstring})
++
++    else ()
++        message("dependency ${pkgname} has no conan package")
++        set(${key}_FOUND FALSE CACHE BOOL ${docstring})
++    endif()
++
++endfunction()
+diff --git a/frmts/hfa/CMakeLists.txt b/frmts/hfa/CMakeLists.txt
+index e5b7138e91..039cac7361 100644
+--- a/frmts/hfa/CMakeLists.txt
++++ b/frmts/hfa/CMakeLists.txt
+@@ -15,7 +15,8 @@ add_gdal_driver(
+           hfa_overviews.cpp
+   BUILTIN)
+ gdal_standard_includes(gdal_HFA)
+-target_include_directories(gdal_HFA PRIVATE $<TARGET_PROPERTY:PROJ::proj,INTERFACE_INCLUDE_DIRECTORIES>)
++target_link_libraries(gdal_HFA INTERFACE PROJ::proj)
++target_include_directories(gdal_HFA PRIVATE ${PROJ_INCLUDE_DIRS})
+ target_compile_definitions(gdal_HFA PRIVATE $<TARGET_PROPERTY:PROJ::proj,INTERFACE_COMPILE_DEFINITIONS>)
+ 
+ if (CMAKE_VERSION VERSION_GREATER_EQUAL 3.12)
+diff --git a/gdal.cmake b/gdal.cmake
+index ff1ca7e6f6..e98875f1b9 100644
+--- a/gdal.cmake
++++ b/gdal.cmake
```

Comment:
For future PR, split the patch file, one by topic. Big patch files are hard to review and probably incompatible to future versions.